cpp/benchmarks/io/parquet/parquet_reader_options.cpp

```diff
@@ -0,0 +1,131 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmarks/common/generate_input.hpp>
+#include <benchmarks/fixture/benchmark_fixture.hpp>
+#include <benchmarks/fixture/rmm_pool_raii.hpp>
+#include <benchmarks/io/cuio_common.hpp>
+#include <benchmarks/io/nvbench_helpers.hpp>
+
+#include <cudf/io/parquet.hpp>
+#include <cudf/utilities/default_stream.hpp>
+
+#include <nvbench/nvbench.cuh>
+
+constexpr size_t data_size = 512 << 20;
+
+std::vector<std::string> get_col_names(cudf::io::source_info const& source)
+{
+  cudf::io::parquet_reader_options const read_options =
+    cudf::io::parquet_reader_options::builder(source);
+  return cudf::io::read_parquet(read_options).metadata.column_names;
+}
+
+template <column_selection ColSelection, row_selection RowSelection, cudf::type_id Timestamp>
+void BM_parquet_read_options(nvbench::state& state,
+                             nvbench::type_list<nvbench::enum_type<ColSelection>,
+                                                nvbench::enum_type<RowSelection>,
+                                                nvbench::enum_type<Timestamp>>)
+{
+  cudf::rmm_pool_raii rmm_pool;
+
+  auto constexpr num_chunks          = 1;
+  auto constexpr str_to_categories   = true;
+  auto constexpr use_pandas_metadata = true;
+
+  auto const ts_type = cudf::data_type{Timestamp};
+
+  // No nested types here, because of https://github.com/rapidsai/cudf/issues/9970
+  auto const data_types =
+    dtypes_for_column_selection(get_type_or_group({static_cast<int32_t>(data_type::INTEGRAL),
+                                                   static_cast<int32_t>(data_type::FLOAT),
+                                                   static_cast<int32_t>(data_type::DECIMAL),
+                                                   static_cast<int32_t>(data_type::TIMESTAMP),
+                                                   static_cast<int32_t>(data_type::DURATION),
+                                                   static_cast<int32_t>(data_type::STRING)}),
+                                ColSelection);
+  auto const tbl  = create_random_table(data_types, table_size_bytes{data_size});
+  auto const view = tbl->view();
+
+  cuio_source_sink_pair source_sink(io_type::HOST_BUFFER);
+  cudf::io::parquet_writer_options options =
+    cudf::io::parquet_writer_options::builder(source_sink.make_sink_info(), view);
+  cudf::io::write_parquet(options);
+
+  auto const cols_to_read =
+    select_column_names(get_col_names(source_sink.make_source_info()), ColSelection);
+  cudf::io::parquet_reader_options read_options =
+    cudf::io::parquet_reader_options::builder(source_sink.make_source_info())
+      .columns(cols_to_read)
+      .convert_strings_to_categories(str_to_categories)
+      .use_pandas_metadata(use_pandas_metadata)
+      .timestamp_type(ts_type);
+
+  auto const num_row_groups = data_size / (128 << 20);
```

Comment:
Should 128 << 20 be a constexpr somewhere?

Reply:
Worse, it should be removed :) This way to calculating the number of row groups is extremely flaky, and that's my this method of row selection is disabled in current benchmarks. My hope is to add `read_parquet_metadata` to properly get the number of row groups and re-enable these benchmark cases.

Reply:
Made `128 << 20` a `constexpr` and added TODO for `read_parquet_metadata` work.